File: ruby/verify.rb
require 'jwt'
require 'openssl'
require_relative 'configurator'

input = ARGV.last
config = Configurator.create_default_configurations
claims = config['claims']
signature_headers = config['signature_headers']
public_key = config.public_key

begin
  JWT.decode(input, public_key, true, { 
    :algorithm => 'RS256',
    :iss => claims['iss'],
    :verify_iss => true,
    :aud => claims['aud'],
    :verify_aud => true,
    'sub' => claims['sub'], 
    :verify_sub => true,
    :verify_iat => true
  })
rescue => error
  print "fail: #{error.message}"
  exit(1)
end
print 'ok'
exit(0)

File: php/verify.php
<?php

require_once __DIR__ . '/Configurator.php';
require_once __DIR__ . '/Checker/IssuerChecker.php';
require_once __DIR__ . '/Checker/SubjectChecker.php';

$config = Configurator::createDefaultConfigurations();

$supported_signatures_algorithms = $config
  ->getOption('supported_signatures_algorithms');
$claims = $config->getOption('claims');
$signature_key = $config->getPublicKey();

$verifier = Jose\Verifier::createVerifier(
  $supported_signatures_algorithms);

$input = end($argv);

$manager = new Jose\Checker\CheckerManager();

$checkers = [];
$checkers[] = new Jose\Checker\AudienceChecker($claims['aud']);
$checkers[] = new Jose\Checker\IssuedAtChecker();
$checkers[] = new Jose\Checker\NotBeforeChecker();
$checkers[] = new Jose\Checker\ExpirationTimeChecker();
$checkers[] = new IssuerChecker($claims['iss']);
$checkers[] = new SubjectChecker($claims['sub']);

foreach ($checkers as $checker) {
  $manager->addClaimChecker($checker);
}

$headerCheckers = [];
$headerCheckers[] = new Jose\Checker\CriticalHeaderChecker();

foreach ($headerCheckers as $checker) {
  $manager->addHeaderChecker($checker);
}

$signature_keyset = new Jose\Object\JWKSet();
$signature_keyset->addKey($signature_key);

$loader = new Jose\JWTLoader($manager, $verifier);

try {
  $jws = $loader->load($input);
  $loader->verify($jws, $signature_keyset);
} catch (Exception $e) {
  print "Failed with " . $e->getMessage();
  exit(1);
}

print 'ok';
exit(0);
